src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -421,7 +360,16 @@ public void RemoveNodeAddress(IPEndPoint ipEndpoint)
             this.logger.LogTrace("({0}:'{1}')", nameof(ipEndpoint), ipEndpoint);
 
             INetworkPeer peer = this.connectedPeers.FindByEndpoint(ipEndpoint);
-            peer?.Disconnect("Requested by user");
+
+            if (peer != null)
+            {
+                peer.Disconnect("Requested by user");
+                this.RemoveConnectedPeer(peer, "Requested by user");
+            }
+
+            this.peerAddressManager.RemovePeer(ipEndpoint.MapToIpv6(), IPAddress.Loopback);
+            List<IPEndPoint> matchingAddNodes = this.ConnectionSettings.AddNode.Where(p => p.Match(ipEndpoint)).ToList();
```

Comment:
Why this funny linq you are enumerating several times.

Also do we need to to the same for -connet? (I am not sure just asking)
```
this.ConnectionSettings.Connect
```

Reply:
I think it is not needed, Connect seems to not be altered when we call AddNodeAddress, however AddNode is modified
![image](https://user-images.githubusercontent.com/4638821/43130210-14be3050-8f2e-11e8-98eb-74fcad5ff2d6.png)